pkg/cmd/cli/cmd/set/env.go

```diff
@@ -139,6 +151,68 @@ func RunEnv(f *clientcmd.Factory, in io.Reader, out io.Writer, cmd *cobra.Comman
 		return err
 	}
 
+	var sources []string
+	if secretName != "" {
+		sources = append(sources, fmt.Sprintf("secret/%s", secretName))
+	}
+	if configMapName != "" {
+		sources = append(sources, fmt.Sprintf("configmap/%s", configMapName))
+	}
+
+	if len(sources) != 0 {
+		mapper, typer := f.Object(false)
+		b := resource.NewBuilder(mapper, typer, resource.ClientMapperFunc(f.ClientForMapping), kapi.Codecs.UniversalDecoder()).
+			ContinueOnError().
+			NamespaceParam(cmdNamespace).DefaultNamespace().
+			FilenameParam(explicit, false, filenames...).
+			SelectorParam(selector).
+			ResourceTypeOrNameArgs(all, sources...).
+			Flatten()
+
+		one := false
+		infos, err := b.Do().IntoSingular(&one).Infos()
```

Comment:
just double checking that this doesn't actually request the object from the API, just resolves the resource and resource name, right?